hexa/workspaces/models.py

```diff
@@ -140,8 +143,15 @@ def update_if_has_perm(self, *, principal: User, **kwargs):
     def delete_if_has_perm(self, *, principal: User):
         if not principal.has_perm("workspaces.delete_workspace", self):
             raise PermissionDenied
+
         delete_database(self.db_name)
-        self.delete()
```

Comment:
Why did you remove `self.delete()` here? My understanding is that we won't use `delete_if_has_perm` for now. But if you leave the method, then it makes sens that it still calls `self.delete()`

Reply:
It shouldn't be removed. I'll put it back